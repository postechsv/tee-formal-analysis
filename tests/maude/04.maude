set show timing off .
set show advisories off .

select NAT .

red s 0 + s s 0 .
red s 0 + s 0 .
red _+_(s 0, N:Nat, s 0) .
red sd(s s 0, s s s 0) .
red sd(s_^3(0), s_^10(0)) .
red sd(s_^10(0), s_^4(0)) .
red s 0 * s s 0 .
red s 0 * s 0 .
red s_^10(0) quo s_^4(0) .
red s_^10(0) rem s_^4(0) .
red s_^10(0) ^ s_^4(0) .