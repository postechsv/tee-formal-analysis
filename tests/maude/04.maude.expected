==========================================
reduce in NAT : 1 + 2 .
rewrites: 1
result NzNat: 3
==========================================
reduce in NAT : 1 + 1 .
rewrites: 1
result NzNat: 2
==========================================
reduce in NAT : N:Nat + 1 + 1 .
rewrites: 1
result NzNat: N:Nat + 2
==========================================
reduce in NAT : sd(2, 3) .
rewrites: 1
result NzNat: 1
==========================================
reduce in NAT : sd(3, 10) .
rewrites: 1
result NzNat: 7
==========================================
reduce in NAT : sd(4, 10) .
rewrites: 1
result NzNat: 6
==========================================
reduce in NAT : 1 * 2 .
rewrites: 1
result NzNat: 2
==========================================
reduce in NAT : 1 * 1 .
rewrites: 1
result NzNat: 1
==========================================
reduce in NAT : 10 quo 4 .
rewrites: 1
result NzNat: 2
==========================================
reduce in NAT : 10 rem 4 .
rewrites: 1
result NzNat: 2
==========================================
reduce in NAT : 10 ^ 4 .
rewrites: 1
result NzNat: 10000
Bye.
